apps/api/src/domain/services/query-services/edge-connection-query.service.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    CategorizableType,
+    ICategorizableIndexQueryResult,
+    INoteViewModel,
+    WithTags,
+} from '@coscrad/api-interfaces';
+import { Inject } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { CommandInfoService } from '../../../app/controllers/command/services/command-info-service';
+import { mixLinkIntoViewModelDescription } from '../../../app/controllers/utilities';
+import mixTagsIntoViewModel from '../../../app/controllers/utilities/mixTagsIntoViewModel';
+import { InternalError } from '../../../lib/errors/InternalError';
+import { RepositoryProvider } from '../../../persistence/repositories/repository.provider';
+import { NoteViewModel } from '../../../view-models/edgeConnectionViewModels/note.view-model';
+import { buildAllAggregateDescriptions } from '../../../view-models/resourceDescriptions';
+import { EdgeConnection } from '../../models/context/edge-connection.entity';
+import { validAggregateOrThrow } from '../../models/shared/functional';
+import { AggregateType } from '../../types/AggregateType';
+import { isNullOrUndefined } from '../../utilities/validation/is-null-or-undefined';
+
+export class EdgeConnectionQueryService {
+    // TODO Program to the interface `IRepositoryProvider`, not the concrete type
+    constructor(
+        @Inject(RepositoryProvider) private readonly repositoryProvider: RepositoryProvider,
+        @Inject(CommandInfoService) private readonly commandInfoService: CommandInfoService,
+        @Inject(ConfigService) private readonly configService: ConfigService
+    ) {}
+
+    async fetchSchema() {
+        const searchResult = buildAllAggregateDescriptions().find(
+            ({ type }) => type === AggregateType.note
+        );
+
+        if (isNullOrUndefined(searchResult)) {
+            throw new InternalError(`Failed to find a view model description for the note model`);
+        }
+
+        const result = mixLinkIntoViewModelDescription(
+            this.configService.get<string>('GLOBAL_PREFIX')
+        )(searchResult);
+
+        return result;
+    }
+
+    /**
+     * In the future, we may want to use Access Control Lists on notes as well.
+     */
+    async fetchMany(): Promise<ICategorizableIndexQueryResult<INoteViewModel>> {
+        const queryResult = await this.repositoryProvider.getEdgeConnectionRepository().fetchMany();
+
+        const validDomainModels = queryResult.filter(validAggregateOrThrow);
```

Comment:
So here this will throw if any of the results returns an error?

Reply:
Yes. It's  a system error if invalid edge-connection DTOs get into our database. We should be notified by the user and fix the issue. Most likely a spoofed migration or maybe introduction of stricter invariant validation rules. This is why I'd like to have a way to validate the entire state (as we do with `testData.spec.ts` post-migration BTW.